recipes/whereami/all/conanfile.py

```diff
@@ -0,0 +1,91 @@
+
+required_conan_version = ">=1.33.0"
+
+
+class WhereamiConan(ConanFile):
+    name = "whereami"
+    description = "Locate the current executable and the current module/library on the file system"
+    topics = ("conan", "whereami", "introspection", "getmodulefilename",
+              "dladdr", "executable-path", "getexecutablepath")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/gpakosz/whereami"
+    license = ("MIT", "WTFPL")
+    exports_sources = ["CMakeLists.txt"]
+    generators = "cmake"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
+                  destination=self._source_subfolder, strip_root=True)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+
+        self._cmake = CMake(self)
+        self._cmake.configure()
+        return self._cmake
+
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy(pattern="LICENSE.*", dst="licenses",
+                  src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+        self._create_cmake_module_alias_targets(
+            os.path.join(self.package_folder, self._module_file_rel_path),
+            {"whereami": "whereami::whereami"}
+        )
+
+    @staticmethod
+    def _create_cmake_module_alias_targets(module_file, targets):
+        content = ""
+        for alias, aliased in targets.items():
+            content += textwrap.dedent("""\
+                if(TARGET {aliased} AND NOT TARGET {alias})
+                    add_library({alias} INTERFACE IMPORTED)
+                    set_property(TARGET {alias} PROPERTY INTERFACE_LINK_LIBRARIES {aliased})
+                endif()
+            """.format(alias=alias, aliased=aliased))
+        tools.save(module_file, content)
+
+    @property
+    def _module_subfolder(self):
+        return os.path.join("lib", "cmake")
+
+    @property
+    def _module_file_rel_path(self):
+        return os.path.join(self._module_subfolder,
+                            "conan-official-{}-targets.cmake".format(self.name))
+
+    def package_info(self):
+        self.cpp_info.libs = tools.collect_libs(self)
+        self.cpp_info.names["cmake_find_package"] = "whereami"
+        self.cpp_info.names["cmake_find_package_multi"] = "whereami"
```

Comment:
```suggestion
```

Since the package name and `cpp_info.names` are the same, I think these definition are unnecessary.

Reply:
well, it might be technically unnecessary, but some contributors (e.g. @SpaceIm) consider it as a flag that recipe has been explicitly verified for target names being created

Reply:
I have little opinion on the subject, do as you guys please, maintainers should have write access to the branch on  my fork.
The bincrafters recipe had it in so I left it there.

Reply:
@jmarrec It's an idea created after Bincrafters. Unfortunately most of recipes from Bincrafters are outdated now.

Reply:
Well in the first place, there is no official CMake config file for this library, so all the cake_find_package stuff can be remove in package() & package_info().